engine-rest/src/main/java/org/camunda/bpm/engine/rest/sub/history/impl/HistoricProcessInstanceResourceImpl.java

```diff
@@ -43,4 +46,16 @@ public HistoricProcessInstanceDto getHistoricProcessInstance() {
     return HistoricProcessInstanceDto.fromHistoricProcessInstance(instance);
   }
 
+  @Override
+  public void deleteHistoricProcessInstance() {
+    HistoryService historyService = engine.getHistoryService();
+    try {
+      historyService.deleteHistoricProcessInstance(processInstanceId);
+    } catch (AuthorizationException e) {
+      throw e;
+    } catch (ProcessEngineException e) {
+      throw new InvalidRequestException(Status.NOT_FOUND, e, "Historic process instance with id " + processInstanceId + " does not exist");
```

Comment:
> ProcessInstance with the given ID exists (the instance has not yet been moved to history)

The history is written for both running and finished process instances. Meaning: when the process instance is started, the history entry is written right away (in the first TX). 

So for the moment I would not deem this necessary